src/app/cube/user-profile/md5.ts

```diff
@@ -0,0 +1,231 @@
+// Taken from https://github.com/killmenot/webtoolkit.md5
```

Comment:
If a package is available, try to utilize it instead of copying code from an external js file into the project.

But try to avoid installing bulky packages if functionality needed is minimal and can be developed in house.

Also if there is a general function that may/can be used elsewhere in the project, moved to shared instead of housing it in component's directory.

A md5 hash package is available here: https://www.npmjs.com/package/md5